on-chain/lib/aiken/merkle-patricia-forestry.ak

```diff
@@ -136,6 +136,30 @@ pub fn delete(
   MerklePatriciaForestry { root: excluding(key, proof) }
 }
 
+/// **fails** when | the [Proof](#Proof) is invalid
+/// ---            | ---
+/// **fails** when | there's no element in the trie at the given key
+/// ---            | ---
+///
+/// Update an element in the trie with a a new value. This requires a [Proof](#Proof)
+/// of the old element, to ensure its in the list, and a [Proof](#Proof) of the new
+/// element, to re-add it.
+///
+/// Can be thought of as a delete, followed by an insert, but is able to do it with one fewer
+/// membership checks
+pub fn update(
+  self: MerklePatriciaForestry,
+  key: ByteArray,
+  proof: Proof,
+  old_value: ByteArray,
+  new_value: ByteArray,
+) {
+  // TODO: could we do this with a single traversal? `something(key, old_value, new_value, proof) -> (old_root, new_root)`?
```

Comment:
I've done -- at some point -- something similar for including / excluding since only the last step in the traversal really is different. But in the end, there's an extra cost that comes from having more arguments to keep track of and you still need the same number of hashes and prefix manipulation so there's almost no visible benefits on the execution costs in doing so.  And, it obfuscates the code a lot more, so I gave up on the idea eventually.